scripts/helpers/metrics/cyclomatic.js

```diff
@@ -0,0 +1,138 @@
+const TypeScript = require("typescript");
+
+/**
+ * @typedef {import("../coverage").Script} Script
+ */
+
+/**
+ * @typedef {import("../coverage").FunctionCoverage} FunctionCoverage
+ */
+
+/**
+ * @typedef {import("../coverage").BlockCoverage} BlockCoverage
+ */
+
+/**
+ * @param {String} source
+ */
+const createSource = source =>
+  TypeScript.createSourceFile(
+    "anon.ts",
+    source,
+    TypeScript.ScriptTarget.ES2015
+  );
+
+/**
+ * @param {Script} script
+ */
+function totalOperations(script) {
+  let total = 0;
+  if (script.sources.length > 1) {
+    // typescript
+    for (let i = 1, n = script.sources.length; i < n; i++) {
+      total += visit(createSource(script.sources[i].content));
+    }
+  } else {
+    // javascript
+    total += visit(createSource(script.sources[0].content));
+  }
+
+  return total;
+}
+
+/**
+ * @param {TypeScript.Node} node
+ */
+function visit(node, depth = -1) {
+  let total = 0;
+
+  switch (node.kind) {
+    case TypeScript.SyntaxKind.CaseClause:
+      const newNode = /** @type {TypeScript.CaseClause} */ (node);
+      total += newNode.statements.length;
+      break;
+    case TypeScript.SyntaxKind.CatchClause:
+    case TypeScript.SyntaxKind.ConditionalExpression:
+    case TypeScript.SyntaxKind.DoStatement:
+    case TypeScript.SyntaxKind.ForStatement:
+    case TypeScript.SyntaxKind.ForInStatement:
+    case TypeScript.SyntaxKind.ForOfStatement:
+    case TypeScript.SyntaxKind.IfStatement:
+    case TypeScript.SyntaxKind.WhileStatement:
```

Comment:
try-catches? switch? possibly functions? (especially anonymous)

Reply:
probably also if else and else

Reply:
As discussed in person:
1. Try and else do not increase complexity, as they are already part of the control flow. 
2. Anonymous functions are covered
3. If else is covered, in the sense that if is covered. See 1. for else.
4. Switch is covered because each case statement is covered.